doc/doxygen/groups.dox

```diff
@@ -0,0 +1,14 @@
+/**
+
+@defgroup common Common API
+@brief Common functions and structures.
+
+*/
+
+/**
+
+@defgroup lua Lua API
```

Comment:
This looks like it belongs from another project

Reply:
This is a doxygen group definition for all headers in the ``common/lua`` folder. So it's related to this change from the other PR -> https://github.com/scummvm/scummvm/pull/2361/files#diff-30c89c991aba94a200fe79266635efa2

This file is a central dox file to manage groups, for example when there are multiple header files that need a 'parent' group, you would define the group here instead of an arbitrary header file. The file can also be used to manipulate the hierarchy of different groups.

I can move it to PR#2361 if you prefer, but they are really both complementary PRs and none will work properly without the other.